test/core/binary.wast

```diff
@@ -368,6 +368,25 @@
   "integer too large"
 )
 
+;; Function with missing end marker
+(assert_malformed
+  (module binary
+    "\00asm" "\01\00\00\00"
+    "\03\02\01\00"             ;; Function section
```

Comment:
You may want to adjust this to have two versions as well.

Perhaps also add a type section, because the function type now actually points to an undefined type index, which may make most decoders bail out early, before even reaching the intended bug.